### ISSUE TITLE
Feature Request: .mjs (es module) migrations and seeds (no cli)

### ISSUE THREAD

**Issue by JasonWoof** [knex/knex#3850]
# Environment

Knex version: 0.21.1
OS: docker node:latest

I'd like to be able to write my migrations (and seed) files as es modules with the .mjs file extension like I'm doing with all my other javascript files.

I'd like to be able to run these migrations from within node (not via the cli) like this:

    db/migrations/00001-users-table.mjs:
        export const up = export const up = knex => knex.schema.createTable('users'...
        export const down = knex => knex.schema.dropTable('users');

    index.mjs:
        import knex from 'knex';
        const db = app.context.db = knex({client: ..., connection: {...}});
        db.migrate.latest({
            extension: 'mjs',
            directory: 'db/migrations'),
        });

Ditto for seed files.

Thanks!

**Comment by D10221** [comment]
Sorry if I'm missing something :)
But I this works today 
```javascript
import knex from 'knex';
// const db = app.context.db = knex({ client: "sqlite", connection: "./knex-mjs.sqlite" });
const db = knex({
    client: "sqlite3",
    connection: {
        filename: "./knex-mjs.sqlite"
    },
    useNullAsDefault: true
});
db.migrate.latest({
    extension: 'mjs',
    directory: 'db/migrations',
}).then(() => {
    console.log("done")
    process.exit()
}).catch(e => {
    console.error(e);
    process.exit(1)
})
```

**Comment by JasonWoof** [comment]
Thanks! I didn't realize I had to specify the extension of my migration files twice... or maybe the "extension:" key is not needed?

I find it really strange that the `extension:` configuration key to `migrate.latest()` doesn't configure it to use migrations with that extension.

I suggest either:

1. The `extension:` config sets that extension for both reading and writing migration files (so my example could would work as-is).

or

2. A note is added to the docs for `extension:' key here http://knexjs.org/#Migrations-API cautioning that this config isn't used when reading migration files and that you need to use `loadExtensions:` for that.

**Comment by D10221** [comment]
Yes, Its not very intuitive, I think.
Tracking the 'extension' property by usage, Seems to be used to generate stubs only.
There is a note, now, in the docs, on the 'esm' section , where it mentions 'loadExtension', specifically for 'mjs'  files.
You have to explicit because is not on the default file extension list.
While is esm related, mjs is part of the current/standard NodeJs module resolution, 
perhaps it should be added to default list of well known extensions ? 
And maybe a jsdoc entry in the type definition ?

**Comment by sombriks** [comment]
yes, mjs should be part of default extension list. if possible, the hability to specify either mjs or cjs on knex init so migrate creation could benefit from that too.

**Comment by mercmobily** [comment]
This is a duplicate of #5800. Please follow that issue for updates on ESM and TypeScript support for migrations and seeds.

**Comment by mercmobily** [comment]
Closing as duplicate of #5800.
